### DISCUSSION TITLE
New vulnerability within the Rust programming language

### DISCUSSION THREAD

**Discussion by liamf91** [rustdesk/rustdesk#7777] [General]
Good day. Today I have been running through current vulnerabilities on our machines. The following has been found that the current version of RustDesk could be using an older version of the Rust language.

Please see below.

Rust is a programming language.
The Rust standard library fails to properly escape arguments passed to batch files on Microsoft Windows when using the Command API. An attacker could
exploit this vulnerability by passing malicious arguments, resulting in arbitrary shell execution.
Affected Versions:
Rust versions before 1.77.2 on Windows

IMPACT:
Successful exploitation allows an attacker to execute arbitrary code on a targeted system.

SOLUTION:
Customers are advised to upgrade to Rust 1.77.2 (https://www.rust-lang.org/tools/install) or later versions to remediate this vulnerability.

Patch:
Following are links for downloading patches to fix the vulnerabilities:
Rust 1.77.2 or later (https://www.rust-lang.org/tools/install)

RESULTS:
HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\RustDesk DisplayName = RustDesk
HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\RustDesk DisplayVersion = 1.2.3.2

**Comment by rustdesk** [comment]
@Nadahar could you help confirm if 1.77.2 can work on windows 7?
https://github.com/rustdesk/rustdesk/discussions/7503

**Reply by Nadahar** [reply]
Sorry, I have uninstalled/removed my RustDesk development environment since there clearly was no interest in what I had to contribute.

That said, this "vulnerability" seems to be blown completely out of proportions, as usual. The only way to exploit this is if you run `.bat` or `.cmd` files from your code while also using *user provided arguments*. I see this as a very unusual thing to do, while of course not impossible. You should easily be able to tell if RustDesk does this or not. Please note that it doesn't apply to calling other type of files like `.exe` files - and it's yet another reason to make sure to specify the extension if you intend to run e.g. an `.exe` file (so, call `cmd.exe` instead of just `cmd` so that there is no confusion and a `cmd.bat` can potentially be executed): https://flatt.tech/research/posts/batbadbut-you-cant-securely-execute-commands-on-windows/

Since Rust 1.77.2 is the latest version of Rust, I can already guarantee that it does NOT work with Windows 7.

It should also be noted that most programming languages have elected NOT to "patch" this, as can be seen in the above provided link. Many have just updated their documentation (Appendix B).

My guess is that the biggest "problem" BatBadBut will cause is all the false positives by stupid "security analyzers" that only looks at the Rust version and not whether this is actually applicable for the application in question.

**Reply by rustdesk** [reply]
Thank you very much
